src/v/kafka/server/handlers/describe_log_dirs.cc

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/server/handlers/describe_log_dirs.h"
+
+#include "cluster/partition_manager.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/util/log.hh>
+
+#include <fmt/ostream.h>
+
+namespace kafka {
+
+using partition_dir_set
+  = absl::flat_hash_map<model::topic, std::vector<describe_log_dirs_partition>>;
+
+static describe_log_dirs_partition describe_partition(cluster::partition& p) {
+    // offset_lag is calculated in kafka as std::max(highwatermark -
+    // log_end_offset, 0) which to me looks like it should always be 0 given
```

Comment:
according to this LEO may be higher that highwatermark and in our case it seems it's the difference between `_visibility_upper_bound_index` (I think it's the same as `dirty_offset`) and `_majority_replicated_index` (majority of dirty offsets) //cc @mmaslankaprv

Reply:
@rystsov yes when LEO is >= high-water mark then `std::max(hwm - leo, 0)` will be zero. on the other hand, how would hwm ever be larger than leo to give a value > 0?

Reply:
here is the line from kip-113

>   offset_lag => int64  // If this is not a temporary replica, then offset_lag = max(0, HW - LEO). Otherwise, offset_lag = primary Replica's LEO - temporary Replica's LEO

I'm not too sure this will ever effect us I think its involved with some kafka-specifc operation, but some users want the reported log dir size bytes. in any case, it'd be nice to understand this more. if we can figure out the goal of this field is maybe we can compute a meaningful value